### PR TITLE
Les 290 Add possibility to change theme from light to dark by clicking button

### PR DESCRIPTION
**Pull request by DimitarShemet**
By clicking on the button we can change themes.

### CODE REVIEW

**Review by valer-sky — APPROVED** [review]

File: src/App.scss [hunk 1]
```diff
@@ -1,16 +1,25 @@
+@import "./components/styles/light-dark-theme";
+body {
+  background-color: var(--bgThemeColor1);
+  color: var(--textThemeColor1);
+  text-align: center;
+  transition: background-color 0.5s ease;
+}
+a {
+  color: var(--textThemeColor1);
```
Comment (Evgeniy121212) [inline src/App.scss:9]: try to use scss variables

like:

`$textThemeColor1`

File: src/components/body/footer/Footer.tsx [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
-  onLanguageSwitch: (arg: string) => void
-}
+  onLanguageSwitch: (arg: string) => void;
+};
 const Footer = (props: FooterProps) => {
   const { onLanguageSwitch } = props;
 
   return (
-    <div style={{display:"flex"}}>
+    <div style={{ display: "flex", justifyContent: "center" }}>
```
Comment (Evgeniy121212) [inline src/components/body/footer/Footer.tsx:13]: please, move it to the styles (not use inline styles)

File: src/components/styles/_light-dark-theme.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  ),
+  bgThemeColor2: (
+    darkTheme: $blue600,
+    lightTheme: $grey200,
+  ),
+  textThemeColor1: (
+    darkTheme: $grey100,
+    lightTheme: $black,
+  ),
+);
```
Comment (Evgeniy121212) [inline src/components/styles/_light-dark-theme.scss:16]: duplicate lines of code (same like into the `src/components/styles/_colors.scss`)

**Review by Evgeniy121212 — COMMENTED** [review]
Please, return `.prettierrc.json`-file back - do not delete or modify it

File: src/App.tsx [hunk 4]
```diff
@@ -1,39 +1,57 @@
...
+              <Route
+                path="/users/sign_in/phone_number/code"
+                element={<CodePage />}
+              />
+              <Route path="/search" element={<Search />} />
+            </Routes>
+          </div>
         </div>
-        <div style={{display: "flex", flexDirection: "column"}}>
+        <div style={{ display: "flex", flexDirection: "column" }}>
```
Comment (Evgeniy121212) [inline src/App.tsx:54]: to the className (no inline style)

File: src/App.scss (hunk 1 above)
Comment (DimitarShemet) [reply src/App.scss:9]: We decided to leave css variables in this place

**Review by Evgeniy121212 — APPROVED** [review]
approving ahead

File: src/utils/useTheme.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import useDarkMode from "use-dark-mode";
+
+const lightTheme = "light-mode";
+const darkTheme = "dark-mode";
+
+export const useTheme = () => {
+  const darkMode = useDarkMode();
+  const [theme, setTheme] = React.useState(darkTheme);
+  React.useEffect(() => {
+    setTheme(darkMode?.value ? darkTheme : lightTheme);
```
Comment (Terapool) [inline src/utils/useTheme.tsx:11]: setTheme(darkMode?.value ? THEME.DARK : THEME.LIGHT);

File: src/utils/useTheme.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  }, [darkMode.value]);
+
+  return theme;
+};
+export const THEME = { DARK: "dark-mode", LIGHT: "light-mode" };
```
Comment (Terapool) [inline src/utils/useTheme.tsx:16]: Please, move this const just after imports

**Review by Terapool — APPROVED** [review]

**Pull request merged**